packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -1,14 +1,22 @@
-import { GraphQLClient } from 'graphql-request';
-import chalk from 'chalk';
+import { GraphQLClient, ClientError } from 'graphql-request';
 import { DocumentNode } from 'graphql';
+import { debugHttp as debug } from './debug';
 
 export class GraphQLRequestClient {
+  private client: GraphQLClient;
+  private headers: Record<string, string> = {};
+
   /**
    * Provides ability to execute graphql query using given `endpoint`
    * @param {string} endpoint The Graphql endpoint
    * @param {string} [apiKey] The API key to use for authentication. This will be added as an 'sc_apikey' header.
    */
-  constructor(private endpoint: string, private apiKey?: string) {}
+  constructor(private endpoint: string, apiKey?: string) {
+    if (apiKey) {
+      this.headers.sc_apikey = apiKey;
+    }
+    this.client = new GraphQLClient(endpoint, { headers: this.headers });
```

Comment:
More of a note... I remember asking why we're initializing a new GraphQLClient for every request and whether it's safe to keep reusing the same instance across multiple requests. The answer was that this was indeed intentional based on documentation, and investigating this further is mentioned in 469713.

Reply:
I think it still makes sense to create a new one here, but definitely a good question for users of this client.